src/remediations/fifi_2.js

```diff
@@ -176,16 +176,38 @@ async function dispatchWorkRequests (workRequests, playbook_run_id) {
         const response = await dispatcher.postV2PlaybookRunRequests(workRequests);
         probes.JobDispatched(response, playbook_run_id);
 
-        // handle aggregate responses
+        // Store all dispatcher runs
+        const dispatcherRuns = response
+            .filter(r => r.id)
+            .map(r => ({
+                dispatcher_run_id: r.id,
+                remediations_run_id: playbook_run_id,
+                pd_response_code: r.code
+            }));
+
+        if (dispatcherRuns.length) {
+            await queries.insertDispatcherRuns(dispatcherRuns);
+        }
+
+        // Check for failures and throw appropriate error
         const results = [... new Set(_.map(response, 'code'))];
-        if (results.some(i=> i >= 300)) {
-            // TODO: we should log some details and maybe return better HTTP status here
-            return null;
+        const failedCodes = results.filter(code => code >= 300);
+        
+        if (failedCodes.length > 0) {
+            log.warn({results}, 'Some dispatcher requests failed');
+
+            throw new errors.BadRequest('DISPATCHER_FAILURES', 'Some dispatcher requests failed');
```

Comment:
Throwing a more specific error if some requests fail so that it's easier to figure out what's going on. We're also saving the error codes in the database so we can see which ones actually failed too.